lib/libzfs/libzfs_dataset.c

```diff
@@ -3809,10 +3809,10 @@ zfs_create(libzfs_handle_t *hdl, const char *path, zfs_type_t type,
 			    "no such parent '%s'"), parent);
 			return (zfs_error(hdl, EZFS_NOENT, errbuf));
 
-		case EINVAL:
+		case ZFS_ERR_WRONG_PARENT:
 			zfs_error_aux(hdl, dgettext(TEXT_DOMAIN,
 			    "parent '%s' is not a filesystem"), parent);
-			return (zfs_error(hdl, EZFS_BADTYPE, errbuf));
+			return (zfs_error(hdl, EZFS_WRONG_PARENT, errbuf));
```

Comment:
I think we can move this processing to zfs_standard_error_fmt(), since it should always mean the same thing.

Reply:
I believe you are right; this piece of code was only preventing a core dump since `zfs_standard_error_fmt()` does not handle EINVAL but now it should be no longer necessary.